osu.Game/Overlays/Chat/ExternalLinkDialog.cs

```diff
@@ -29,5 +28,13 @@ public ExternalLinkDialog(string url, Action openExternalLinkAction)
                 },
             };
         }
+        protected Boolean isOsuWebSite(string url)
```

Comment:
1. `return url.StartsWith("http://osu.ppy.sh") ||  url.StartsWith("https://osu.ppy.sh") || url.StartsWith("osu.ppy.sh")`
2. Can be inlined.

Reply:
Treating a url as a string is error-prone and potentially dangerous and should use Uri class anyway.

Reply:
You are right, @UselessToucan . Going on a real white-list (configured within the settings for example) would be better. I'll try to go for this :)

About treating the url as a string...  @ekrctb  I'm not totally sure what's the problem since ExternalLinkDialog method was already using the url as a string, I just added a little check. 
But you are likely right that it's not the best approach

Reply:
Actually, I tried a few things but I'm not very confident with the ConfigManager and the way to import the whitelist between files. 
On the other hand, I'm not 100% sure what @ekrctb meant about the url that was used as a String.
So I thought importing the whitelist from the OsuConfigManager to the ExternalLinkOpener file to check wheter the url starts with any domain of the whitelist or not... But this would still consider the url as a String (as it was initially in the merged ExternalLinkOpener) which seems to be wrong.

That's the only approach I have in mind but it looks not right to @ekrctb :/ 
It would be very helpful to me to have a very short explanation about the problem so I could improve it :/

Reply:
If you are doing `s.StartsWith("osu.ppy.sh")`, it accepts "osu.ppy.sh.evil.example.com" as trusted and this is bad.
I haven't read relevant codes closely so I may be wrong but,
First of all, if you are matching for whitelisted domain, then, you have to extract domain from the url right? In order to do so, you probably want to parse the url string with `Uri` constructor then get the information rather than doing string matching for example.
And then, I think it is important (for security reason in general, I'm not sure for this particular case but it is good to be defensive) to stringify `Uri` instance to get a canonical uri string and forget the original string (use only `Uri` instance and its `toString` result) because multiple url parsers can have different interpretation for non-canonical string and this difference can be a problem.
> ExternalLinkDialog method was already defining the url as a string ...

And yes, You may want to this on upper level of function calling hierarchy.